app/models/pageflow/image_file.rb

```diff
@@ -1,84 +1,63 @@
-    include ImageFileStateMachine
-    include UploadedFile
+    include HostedFile
+    include ProcessedFileStateMachine
 
-    STYLES = lambda do |attachment|
+    def attachment_styles(attachment)
       panorama_format = File.extname(attachment.original_filename) == '.png' ? :PNG : :JPG
 
       Pageflow
         .config.thumbnail_styles
-        .merge(print: ['300x300>', :JPG],
-               medium: ['1024x1024>', :JPG],
-               large: ['1920x1920>', :JPG],
-               ultra: ['3840x3840>', :JPG],
-               panorama_medium: ['1024x1024^', panorama_format],
-               panorama_large: ['1920x1080^', panorama_format])
+        .merge(
+          print: {geometry: '300x300>',
+                  format: :JPG,
+                  convert_options: '-quality 10 -interlace Plane'},
+          medium: {geometry: '1024x1024>',
+                   format: :JPG,
+                   convert_options: '-quality 70 -interlace Plane'},
+          large: {geometry: '1920x1920>',
+                  format: :JPG,
+                  convert_options: '-quality 70 -interlace Plane'},
+          ultra: {geometry: '3840x3840>',
+                  format: :JPG,
+                  convert_options: '-quality 90 -interlace Plane'},
+          panorama_medium: {geometry: '1024x1024^',
+                            format: panorama_format,
+                            convert_options: '-quality 90 -interlace Plane'},
+          panorama_large: {geometry: '1920x1080^',
+                           format: panorama_format,
+                           convert_options: '-quality 90 -interlace Plane'}
+        )
     end
 
-    CONVERT_OPTIONS = {
-      print: '-quality 10 -interlace Plane',
-      medium: '-quality 70 -interlace Plane',
-      large: '-quality 70 -interlace Plane',
-      ultra: '-quality 90 -interlace Plane',
-      panorama_medium: '-quality 90 -interlace Plane',
-      panorama_large: '-quality 90 -interlace Plane'
-    }.freeze
-
-    has_attached_file(:unprocessed_attachment,
-                      Pageflow.config.paperclip_s3_default_options)
-
-    has_attached_file(:processed_attachment,
-                      Pageflow.config.paperclip_s3_default_options
-                        .merge(default_url: ':pageflow_placeholder',
-                               styles: STYLES,
-                               convert_options: CONVERT_OPTIONS))
-
-    validates :attachment, presence: true
-
-    do_not_validate_attachment_file_type(:unprocessed_attachment)
-    do_not_validate_attachment_file_type(:processed_attachment)
-
-    after_unprocessed_attachment_post_process :save_image_dimensions
-
-    def attachment
-      processed_attachment.present? ? processed_attachment : unprocessed_attachment
+    def attachment_default_url
+      ':pageflow_placeholder'
     end
 
-    def attachment=(value)
-      self.unprocessed_attachment = value
-    end
+    do_not_validate_attachment_file_type(:attachment)
```

Comment:
Should no longer be needed, right?

Reply:
Removed now and replaced by filename validation (disallowing forward and backward slashes in the filename)

Reply:
I mean, not required _here_ since `HostedFile` already defines it.